internal/apex/apex.go

```diff
@@ -403,10 +405,6 @@ func (ax *Apex) relayStateReconcile(zoneID uuid.UUID) error {
 	return nil
 }
 
-func (ax *Apex) Shutdown(ctx context.Context) error {
-	return nil
-}
```

Comment:
Could you take a look at this PR?  https://github.com/redhat-et/apex/pull/341

I was using Apex.Shutdown() in there. I want to make sure I understand how to adjust that PR to work with this. It looks like I could select on `<-ctx.Done()` instead of the custom channel I had created to write to from Apex.Shutdown() ?

Reply:
I'm pretty sure I've answered my own question, haha. Feel free to comment here or on my PR if I've missed something. I'll review this again later today.

Reply:
If you want I can rebase on top of your PR

Reply:
No, I think I'd rather merge this first. I like what you've done and it'll simplify mine a little bit.